src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitions.cs

```diff
@@ -155,5 +161,124 @@ private bool CheckReturnType(IdentToken returnType, List<TexlError> errors)
 
             return isReturnTypeCheckSuccessful;
         }
+
+// This code is intended as a prototype of the Partial attribute system, for use in solution layering cases
+// Provides order-independent ways of merging named formulas
+        #region Partial Attributes
+
+        private static readonly string _renamedFormulaGuid = Guid.NewGuid().ToString("N");
+
+        /// <summary>
+        /// For NamedFormulas with partial attributes,
+        /// validates that the same attribute is applied to all matching names,
+        /// then applies name mangling to all, and constructs a separate
+        /// formula with the operation applied and the original name.
+        /// </summary>
+        private ParseUserDefinitionResult ProcessPartialAttributes(ParseUserDefinitionResult parsed)
+        {
+            var groupedFormulas = parsed.NamedFormulas.GroupBy(nf => nf.Ident.Name.Value);
+            var errors = parsed.Errors?.ToList() ?? new List<TexlError>();
+            var newFormulas = new List<NamedFormula>();
+
+            foreach (var nameGroup in groupedFormulas)
+            {
+                var name = nameGroup.Key;
+                var firstAttribute = nameGroup.Select(nf => nf.Attribute).FirstOrDefault(att => att != null);
+
+                if (firstAttribute == null || nameGroup.Count() == 1)
+                {
+                    newFormulas.AddRange(nameGroup);
+                    continue;
+                }
+
+                var updatedGroupFormulas = new List<NamedFormula>();
+                var id = 0;
+                foreach (var formula in nameGroup)
+                {
+                    // This is just for the prototype, since we only have the one kind.
+                    if (formula.Attribute.AttributeName.Name != "Partial")
+                    {
+                        errors.Add(new TexlError(formula.Attribute.AttributeOperationToken, DocumentErrorSeverity.Severe, TexlStrings.ErrOnlyPartialAttribute));
+                        continue;
+                    }
+
+                    if (!firstAttribute.SameAttribute(formula.Attribute))
+                    {
+                        errors.Add(new TexlError(formula.Attribute.AttributeOperationToken, DocumentErrorSeverity.Severe, TexlStrings.ErrOperationDoesntMatch));
+                        continue;
+                    }
+
+                    var newName = new IdentToken(name + _renamedFormulaGuid + id, formula.Ident.Span, isNonSourceIdentToken: true);
+                    id++;
+                    updatedGroupFormulas.Add(new NamedFormula(newName, formula.Formula, formula.StartingIndex, formula.Attribute));
+                }
```

Comment:
You can skip this foreach if 
firstAttribute.AttributeOperation == PartialAttribute.AttributeOperationKind.Error

Reply:
Yes, that was how it was set up previously, but as your earlier comment indicated, that was confusing. Also, doing it this way, we can correctly mark multiple errors, rather than just the first one.

Reply:
Sure. Sorry I can't see last iteration here.